sdk/core/azure-core/src/main/java/com/azure/core/implementation/StringBuilderWriter.java

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Objects;
+
+/**
+ * Implementation of {@link Writer} that write content to a {@link StringBuilder}.
+ */
+public final class StringBuilderWriter extends Writer {
```

Comment:
Is this type required? It seems like it's only needed to track the `closed` status which isn't called from `BinaryDataDeserializer`.

Reply:
No, this type isn't explicitly required. I made it as there is both `String JsonParser.getText()` and `int JsonParser.getText(Writer)` and from reading the docs the `Writer` based API may reduce overhead (and I verified this as implementations of JsonParser don't need to create a string from a bytes/chars range and rather just do a copy) and I wanted to use a `StringBuilder` to write to but there is not `StringBuilder`-based writer